src/pubsub/mod_pubsub_cache_mnesia.erl

```diff
@@ -0,0 +1,74 @@
+-module(mod_pubsub_cache_mnesia).
+
+-include("pubsub.hrl").
+-include("jlib.hrl").
+
+-export([start/0, stop/0]).
+
+-export([
+         upsert_last_item/5,
+         get_last_item/2,
+         delete_last_item/2]).
+
+%% ------------------------ Backend start/stop ------------------------
+
+-spec start() -> ok.
+start() ->
+    create_table().
+
+-spec stop() -> ok.
+stop() ->
+    ok.
+
+-spec upsert_last_item(ServerHost :: binary(),
+                       Nidx :: mod_pubsub:nodeIdx(),
+                       ItemID :: mod_pubsub:itemId(),
+                       Publisher::jid:ljid(),
+                       Payload::mod_pubsub:payload()) -> ok | {error, Reason :: term()}.
+upsert_last_item(_ServerHost, Nidx, ItemId, Publisher, Payload) ->
+    try mnesia:dirty_write(
+        {pubsub_last_item,
```

Comment:
Creating a record with manual tuple construction is evil.

https://github.com/esl/MongooseIM/blob/master/include/pubsub.hrl#L155